process/smartContract/processorV2/processV2.go

```diff
@@ -1033,91 +1076,124 @@ func (sc *scProcessor) resolveBuiltInFunctions(
 	return vmOutput, nil
 }
 
+type sameShardExecutionDataAfterBuiltIn struct {
+	isSCCallSelfShard    bool
+	callType             vmData.CallType
+	parsedTransfer       *vmcommon.ParsedESDTTransfers
+	scExecuteOutTransfer *vmcommon.OutputTransfer
+	err                  error
+}
+
 func (sc *scProcessor) isSameShardSCExecutionAfterBuiltInFunc(
 	tx data.TransactionHandler,
 	vmInput *vmcommon.ContractCallInput,
 	vmOutput *vmcommon.VMOutput,
-) (bool, vmData.CallType, *vmcommon.ParsedESDTTransfers, *vmcommon.OutputTransfer, error) {
+) *sameShardExecutionDataAfterBuiltIn {
+	noExecutionFound := sameShardExecutionDataAfterBuiltIn{
```

Comment:
Use &sameShardExecutionDataAfterBuiltIn here and return it without & below

Reply:
done